### PR TITLE
Use Sample::operator(i, j)

### PR DESCRIPTION
**Pull request by jschueller**

### CODE REVIEW

File: lib/src/Base/Optim/OptimizationResult.cxx [hunk 1]
```diff
@@ -420,10 +420,10 @@ Graph OptimizationResult::drawOptimalValueHistory() const
   for (UnsignedInteger i = 1; i < size; ++ i)
   {
     const UnsignedInteger j = 0;
-    if (!((minimization && (data[i][j] < data[i - 1][j]))
-          || (!minimization && (data[i][j] > data[i - 1][j]))))
+    if (!((minimization && (data(i, j) < data[i - 1][j]))
+          || (!minimization && (data(i, j) > data[i - 1][j]))))
     {
-      data[i][j] = data[i - 1][j];
+      data(i, j) = data[i - 1][j];
```
Comment (dbarbier) [inline lib/src/Base/Optim/OptimizationResult.cxx]: Missing changes here and just above ;-)

File: lib/src/Base/Stat/CorrelationAnalysis.cxx [hunk 2]
```diff
@@ -99,9 +99,9 @@ Point CorrelationAnalysis::PCC(const Sample & inputSample,
     // Build the truncated sample
     for (UnsignedInteger i = 0; i < size; ++i)
     {
-      for (UnsignedInteger j = 0; j < index; ++j) truncatedInput[i][j] = inputSample[i][j];
-      for (UnsignedInteger j = index + 1; j < dimension; ++j) truncatedInput[i][j - 1] = inputSample[i][j];
```
Comment (dbarbier) [inline lib/src/Base/Stat/CorrelationAnalysis.cxx]: Here too

File: lib/src/Base/Stat/CorrelationAnalysis.cxx [hunk 3]
```diff
@@ -99,9 +99,9 @@ Point CorrelationAnalysis::PCC(const Sample & inputSample,
...
-      remainingInput[i][0] = inputSample[i][index];
```
Comment (dbarbier) [inline lib/src/Base/Stat/CorrelationAnalysis.cxx]: ... and here

**Comment by jschueller** [comment]
I replaced in several other locations where my regex didnt catch everything.

**Comment by regislebrun** [comment]
+1

File: lib/src/Base/Stat/FieldImplementation.cxx [hunk 4]
```diff
@@ -157,7 +157,7 @@ Scalar & FieldImplementation::operator () (const UnsignedInteger i,
   // No copyOnWrite() as the at() method already do it
   return at(i, j);
 #else
-  return (*this)[i][j];
+  return (*this)(i, j);
```
Comment (jschueller) [inline lib/src/Base/Stat/FieldImplementation.cxx]: @dbarbier @regislebrun I introduced an infinite recursion in Field::operator(i,j), I'll fix it quickly

**Pull request merged**